lib/giraph/remote/response.rb

```diff
@@ -0,0 +1,9 @@
+module Giraph
+  module Remote
+    # Dummy Hash-wrapper to enable precise `instance_of?` checks
+    # Allows us to differentiate JSON responses that are
+    # coming from remote GraphQL interface vs regular Hash objects
+    class Response < Hash
```

Comment:
A less invasive way of doing this would be to make this a 'marker interface' and dynamically mix it into the response hash.
```ruby
module Giraph
  module Remote
    # define marker 'interface'
    module Response
      # empty
    end
  end
end

# mix-in to Hash
response = JSON.parse(raw_json, symbolize_names: true).extend(Giraph::Remote::Response)

response.is_a?(Giraph::Remote::Response) #=> true
```